src/renderer/services/bitcoin/ledger.ts

```diff
@@ -18,10 +20,47 @@ const retrieveLedgerAddress = (network: Network) =>
     catchError((error) => Rx.of(RD.failure(error)))
   ).subscribe(setLedgerAddressRD)
 
-const createLedgerService = (): LedgerService => ({
+export const createLedgerService = (): LedgerService => ({
   ledgerAddress$,
   retrieveLedgerAddress,
   removeLedgerAddress: () => setLedgerAddressRD(RD.initial)
 })
 
-export { createLedgerService }
+export const createLedgerTransferService = () => {
+  const { get$: ledgerTxRD$, set: setLedgerTxRD } = observableState<LedgerTxRD>(RD.initial)
+
+  const ledgerTx$ = (params: LedgerTxInfoParams): LedgerTxLD =>
+    FP.pipe(
+      Rx.from(createTxInfo(params)),
+      switchMap((ledgerTxInfo) => window.apiHDWallet.signTxInLedger('BTC', params.network, ledgerTxInfo)),
+      switchMap((signResult) =>
+        Either.isRight(signResult)
+          ? Rx.from(broadcastTx({ txHex: signResult.right, nodeUrl: params.nodeUrl, nodeApiKey: params.nodeApiKey }))
+          : Rx.from(() =>
+              Promise.reject({
+                ledgerErrorId: signResult.left
+              })
+            )
+      ),
+      map(RD.success),
+      startWith(RD.pending),
+      catchError(
+        (e): LedgerTxLD =>
+          Rx.of(
+            RD.failure({
+              msg: !e.ledgerErrorId && e.toString(),
+              errorId: ErrorId.SEND_LEDGER_TX,
+              ledgerErrorId: e.ledgerErrorId
+            })
+          )
+      )
+    )
+
+  const pushLedgerTx = (params: LedgerTxInfoParams): Rx.Subscription => ledgerTx$(params).subscribe(setLedgerTxRD)
+
+  return {
+    ledgerTxRD$,
+    pushLedgerTx,
+    resetLedgerTx: () => setLedgerTxRD(RD.initial)
+  }
+}
```

Comment:
Pls move everything of `createLedgerTransferService` into `createLedgerService`. Update `LedgerService` and remove `createLedgerTransferService`